swarm/pss/prox_test.go

```diff
@@ -133,99 +123,132 @@ func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
+		d.requiredMessages += len(targets)
 		for _, id := range targets {
-			recipients[i] = append(recipients[i], id)
-			expectedMsgs[id] = append(expectedMsgs[id], uint64(i))
+			d.recipients[i] = append(d.recipients[i], id)
+			d.expectedMsgs[id] = append(d.expectedMsgs[id], uint64(i))
 		}
 
-		log.Debug("nn for msg", "targets", len(recipients[i]), "msgidx", i, "msg", common.Bytes2Hex(msgAddr[:8]), "sender", senders[i], "senderpo", smallestPo)
+		log.Debug("nn for msg", "targets", len(d.recipients[i]), "msgidx", i, "msg", common.Bytes2Hex(msgAddr[:8]), "sender", d.senders[i], "senderpo", smallestPo)
 	}
-	log.Debug("msgs to receive", "count", msgsToReceive)
+	log.Debug("msgs to receive", "count", d.requiredMessages)
 }
 
 func TestProxNetwork(t *testing.T) {
-	if (*runNodes > 0 && *runMessages == 0) || (*runMessages > 0 && *runNodes == 0) {
-		t.Fatal("cannot specify only one of flags --nodes and --messages")
-	} else if *runNodes > 0 {
-		t.Run(fmt.Sprintf("%d/%d", *runMessages, *runNodes), testProxNetwork)
-	} else {
-		t.Run("1/4", testProxNetwork)
+	t.Run("16/16", testProxNetwork)
+}
+
+// params in run name: nodes/msgs
+func TestProxNetworkLong(t *testing.T) {
+	if !*longrunning {
+		t.Skip("run with --longrunning flag to run extensive network tests")
 	}
+	t.Run("8/100", testProxNetwork)
+	t.Run("16/100", testProxNetwork)
+	t.Run("32/100", testProxNetwork)
+	t.Run("64/100", testProxNetwork)
 }
 
-// This tests generates a sequenced number of messages with random addresses.
-// It then calculates which nodes in the network have the address of each message
-// within their nearest neighborhood depth, and stores them as recipients.
-// Upon sending the messages, it verifies that the respective message is passed to the message handlers of these recipients.
-// It will fail if a recipient handles a message it should not, or if after propagation not all expected messages are handled (timeout)
+// This tests generates a number of messages with random addresses. Then,
+// for each message it calculates which nodes in the network the msg address
+// within its nearest neighborhood depth, and stores those nodes as possible
+// recipients. Those nodes that are the closest to the message address (nodes
+// belonging to the deepest PO wrt msg address) are stored as required recipients.
+// Upon sending the messages, the test verifies that the respective message is
+// passed to the message handlers of these required recipients. Test will fail
+// if a message is handled by recipient which is not listed among the allowed
+// recipients of this particular message. It also fails after timeout, if not
+// all the required recipients have received their respective messages.
+//
+// For example, if proximity order of certain msg address is 4, and node X
+// has PO=5 wrt the message address, and nodes Y and Z have PO=6, then:
+// nodes Y and Z will be considered required recipients of the msg,
+// whereas nodes X, Y and Z will be allowed recipients.
```

Comment:
We should mention on the top that it tests a specific functionality of the `pss` package (setting the `prox` property of a `handler`)

The last comment here (a very nice explanation, thanks!) should perhaps be in `swarm/pss/pss.go` or `swarm/pss/types.go` itself, instead of here in the test.

Reply:
the last example must be here, otherwise 90% of the readers will not understand it, despite the perfectly correct theoretical explaination.

Reply:
Ok, but we should have this documentation in the implementation file aswell, really. People shouldn't have to read the tests to get the relevant information.